### DISCUSSION TITLE
setFilterValue - Sets the filter but does not put value in the column filter field.

### DISCUSSION THREAD

**Discussion by kscottthomas** [KevinVandy/material-react-table#156] [Q&A]
I am using the following trying to set the value in the column's filter text value see below:

`tableInstanceRef.current.getColumn(id).setFilterValue(value);` 

Below is a snippet of the column definition:

`    const columns = useMemo(
        () => [
            {
                accessorKey: 'FirstName',
                header: 'First Name',
                filterFn: 'startsWith',
            },......`

Oddly this does set the filter for the table but not the value in the text field to reflect the current state. Is this a bug or not supported?

BTW: I am trying to avoid adding the <TextField component into the definition.... 

Thank you in Advance! Scott

**Comment by KevinVandy** [comment]
The built-in filter textfield has its own state for performance and debouncing reasons, so this is kind of a limitation right now.

There might actually be a solution though that is kind of strange. All the filterRefs are also stored on the tableInstance. You might be able to mutate its state manually.

Try something like this:

```ts
tableInstanceRef.current.getColumn(id).setFilterValue(value); //change value
tableInstanceRef.current.refs.filterInputRefs[id].value = value; //change internal input
```

**Reply by kscottthomas** [reply]
Again... thank you for your fast response. I will try your suggestion.

**Reply by kscottthomas** [reply]
Well it almost worked... the syntax was 

`tableInstanceRef.current.refs.filterInputRefs.current[id + '-0'].value = value;`

It did change the value in the built in text field however it was short lived when the final render removed it.

**Reply by KevinVandy** [reply]
This is pretty much a bug/limitation with how it is currently coded. I'll probably be able to get rid of this limitation by not using localized state for the inputs when I start working on version 2.0, which will focus on using React 18 concurrent rendering features.
